### ISSUE TITLE
Fluid Requester station calculates L signal (train limit) wrong

### ISSUE THREAD

**Issue by asdkant** [Opinionated-Blueprints/10-Books-Full-of-Rails#84]
### Have you checked the following resources?

- [X] [Wiki](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/wiki)
- [X] [Discussions Forum](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/discussions)
- [X] [Open and Closed Issues](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/issues)

### Version of the Book

2.19.0

### Bug Description

(for context, I'm trying to do a 2-4-0 crude oil requester and I already have a crude oil provider with a wagon ready for delivery)

Fluid Requester station calculates L signal (train limit) wrong

### How to reproduce the problem

I tried tracing the issue going backwards from the station.

- The station sets train limit to the L signal, get a green wire from the output of the two decider combinators
- the two decider combinators get the max train limit in the green wire and the calculated train limit on the red wire, and pass the smaller one
- that red wire comes from an arithmetic combinator:
  ![image](https://github.com/user-attachments/assets/99edcf4e-4dc5-46b9-a193-fdc74fcb1e37)
  
  this seems to correspond to a fluid *provider*, not to a fluid *requester*

I'm copying the export string of the blueprint below:
```
0eNrdWdtu4zYQ/RVCj1nJa1mSb4D74ADtPqRo0A2SB8cwaIu2idVtKcppEOQD+h/9sn5JZ6iLZYe2rHRR7DYPNk1yzgzJMxcyL8YyyFgieCSN8Yvhs3QleCJ5HBlj4zOTKZFbRqSgPCIBD7kkMlZdURYumSDxmqyzIMhnBDH1UxLyNOXRxiQrmiTMJ1QSStKErfiaw8+Q/sHDLOw8Rgo+EbHg8hlhV9s4ZRHZUbCIfGjW0DFMg6/iKDXGsxcj5ZuIBrgI+ZwwsH7HhcygxzQiGmJHPsNaBxn3rYQK6JVMGK+AEvnsD2Nsv85Ng0WSS85yUPXjeZGbAhMqLApGb0Ger6xVHC55RGUsQFUSpzzfvRcDEC1v0DWNZ2gMer2OB6p8Ltgqn+GaBlgvRRwslmxLdxwQQGwPvYBhX8GlOLDmIpWLi9d5hytLGWLUhIo51fItG6fFCRO0OHULkOJMJplsFKtbtIiYfIrFF2WqYL4xliJjprERjAHsmgYpOzborYyaVgkhwusrSCUBfWZicUjPWbFyKweblOuekwnQSdJgma3XwKBdHGQhe4xmXLJwUlsCTtyzDEdTksI5AkuB7rn0Y5TvRY5I1mBYMYL8piugB+zEG6b0qtNQvLUANtHSwyvpYR+Sw+7p2aHgFggHvtWaDHKh5NOF8uXyiOp9l0PelJAQPZQPl3Dl78uhbg085FSWFDw+qe6c5F2Cfc1Yil2rLXzPiUXenqqBOxfk2wUf4IAb9blUDkyh3XFHw64z6tkefNlu3+n3nK7bHw68V81ZOpW5IfN5FlosgFMS4PpJHDDtqYKvV+fa0YK67wglTonZPQ4lp9jyrWLJrTaWXEKQemT5oIssl2g+H2c0MaMxzmhiU9s4c4vhY0lTRkrClyEGNFhPcDjzcxHqKL35LOA7JiDxQHrlGIfUrkGQiTBH7rjPqozJq5R5qb4PJ9egC15edQg+ZG0fva0hx3UOwpiOiQXSEQ3rv2atWXnzblbCejBi4HrGxt9//mX8Vxyblx7QsmhRVl+ULeeneazIoiafpcsN0hPKtHrZdym1b95L7d/yLDshl1v5GN3FGwbFqCBPEOZUWRqrnwXZSLHV5LSxaxGHb1WSWBy7lkmethySDiyEwBqC+ImJjs55+m2dx/6hneenH9R1tBr+F67TrOf7dJxB6zKrP2wss4YVKF4UJY1kgzvWMd+UWUO9e6b5eHrYBo6W90p0kQCK0+Neza3KNL7Cfqtq2ohiESpCH3jcRHVkeGG3ccUFHFSAjY7wbnDHbK7TWoF73Rq624j+0A69VwP3GsHv25oOfzUF/UYF1+0UWPWdHzSiT9uhw2ZDgFqUv3ojd9QfwGdN57BR56eW54F71qx3ZDZfY9vtpKNX6kCobwr2R+8Evxc3T5+o9yMI6q0yQIsrhH7qA05VqskT3UCAOTnz/mhm9f7xcHWPSUIvdV29mhSWV68m+vlTZXr+EIJvJnTDCITXL3A/wbSCik/KfkLZulDNxOnVp8m5a9OKBqssoHgQB887uowyan/H7o86p15khv/yiq1/QmtbgV0fX6o/vuNSffMdXKp1TpYfpap+FH/3j3FYopQPct+Iwe+ut65LCp6pj3R0tLv7c0ALN1tpwVdw9l3QHh09GuuA7RbAThvgXgtguw2wczlwf3QSGK4EWGNiWTWzTdscmfbchFbPHJg91XKhD1s96HNNt2j1VcsBiYGScCoJx/Sg5UHLhdG+GnVhdKhGvarPU9oc1dqPugVyv+oDfGihxLCwD2yuWJ/Xg4Wzcn/vp7+IOMP3YuirR4yuopVm/s/oLm/n2/X5xd6bZWNs2HuAXyFj3SmvuVEvw2flvO5ecIrZ67Z8AT4r5h7ZSx5UIoPeHQRRugxw5KFJNfxpcch9ngcO0Xbn4azaFtyp0JFvgg4Lq7g1VBzg+oh8BdhAR5YwKF4iWcbIM8pUFVTbuTx9fS7y4B3mQZ3eadManOM1FMgVWN3q6dVUZzewEgMyTNn/P9A0ArpkAQbrNSS+ic/WNAuktYwDfz5TT+wTnD0vayNB1Hmks49qcD77iHL4Hg8xNVU+6/WxBht57sDpdR24sv0DscHGxw==
```

### What did you expect?

L signal would be calculated correctly so trains start delivering

### What happened instead?

L signal was calculated as zero so no train comes to an empty fluid requester station

**Comment by Bocian-1** [comment]
I see you're on 1.19.0. This was fixed on 1.19.1, though I'm seeing now I should clear up the descriptions since they're not entirely accurate

**Comment by Toskin** [comment]
As of v2.21.0, fluid requester won't set up limit because decider sending L signal is set to output it through non-existent red wire instead of connected green wire.
